File: backend/app/services/material_preheat.py
"""Default and bootstrap logic for material preheat presets."""

from sqlalchemy.orm import Session

from app.models.material_preheat_preset import MaterialPreheatPreset

DEFAULT_PRESETS: list[tuple[str, float, float, int]] = [
    ("PLA", 210, 60, 0),
    ("PETG", 240, 80, 1),
    ("ABS", 250, 100, 2),
    ("TPU", 220, 50, 3),
]


def ensure_default_preheat_presets(db: Session) -> list[MaterialPreheatPreset]:
    """Insert factory defaults when the table is empty."""
    existing = db.query(MaterialPreheatPreset).order_by(MaterialPreheatPreset.sort_order).all()
    if existing:
        return existing
    for name, hotend, bed, order in DEFAULT_PRESETS:
        db.add(
            MaterialPreheatPreset(
                name=name,
                hotend_c=hotend,
                bed_c=bed,
                sort_order=order,
            )
        )
    db.commit()
    return db.query(MaterialPreheatPreset).order_by(MaterialPreheatPreset.sort_order).all()

File: frontend/src/types/materialPreheat.ts
export type MaterialPreheatPreset = {
  id: number
  name: string
  hotend_c: number
  bed_c: number
  sort_order: number
}

export const MOCK_PREHEAT_PRESETS: MaterialPreheatPreset[] = [
  { id: 1, name: 'PLA', hotend_c: 210, bed_c: 60, sort_order: 0 },
  { id: 2, name: 'PETG', hotend_c: 240, bed_c: 80, sort_order: 1 },
  { id: 3, name: 'ABS', hotend_c: 250, bed_c: 100, sort_order: 2 },
  { id: 4, name: 'TPU', hotend_c: 220, bed_c: 50, sort_order: 3 },
]
